puppet/zulip/manifests/common.pp

```diff
@@ -32,4 +32,52 @@
   $supervisor_conf_dir = "${supervisor_system_conf_dir}/zulip"
 
   $total_memory_mb = Integer($::memorysize_mb)
+
+  $versions = {
+    # https://github.com/cactus/go-camo/releases
+    'go-camo' => ['2.3.0', {
+      'amd64'   => '965506e6edb9d974c810519d71e847afb7ca69d1d01ae7d8be6d7a91de669c0c',
+      'aarch64' => '40463f6790eb0d2da69ad6a902fcc4c6b0c0ac24106a6c28fbfce9dfa4cb15cd',
+    }],
```

Comment:
Any reason not to use structured hashes instead of two-element arrays?

```puppet
    'go-camo' => {
      'version' => '2.3.0',
      'sha256'  => {
        'amd64'   => '965506e6edb9d974c810519d71e847afb7ca69d1d01ae7d8be6d7a91de669c0c',
        'aarch64' => '40463f6790eb0d2da69ad6a902fcc4c6b0c0ac24106a6c28fbfce9dfa4cb15cd',
      },
    },
```

Reply:
Yeah, no strong one.  Happy to clean that up.

Tangentially, moving this into an external JSON or YAML file is an option; I didn't because it seemed like an extra indirection to have to chase, but Puppet does support doing so.

Reply:
Alternatively, it’d be nice to further centralize third-party versions and hashes to a file that could also be used by `install-node`, `install-yarn`, `install-shellcheck`, `install-shfmt`, `bootstrap-awscli.sh`, and `dev-vagrant-docker`. Perhaps the most useful common format would be a flat list of assignments:

```sh
go_camo_version=2.3.0
go_camo_sha256_amd64=965506e6edb9d974c810519d71e847afb7ca69d1d01ae7d8be6d7a91de669c0c
go_camo_sha256_aarch64=40463f6790eb0d2da69ad6a902fcc4c6b0c0ac24106a6c28fbfce9dfa4cb15cd
```

that could be directly sourced from shell and should be easily parsed by Puppet functions.

Reply:
Yeah, I think something in that direction would be a good idea.  We do already have code to read `.ini` files from puppet, so another direction would be to have a file in that format:
```
[go_camo]
version=2.3.0
sha256_amd64=965506e6edb9d974c810519d71e847afb7ca69d1d01ae7d8be6d7a91de669c0c
...
```

which could be parsed using `crudini` in shell scripts and by puppet functions as well, and might feel a bit nicer as a standard format that can have comments etc.

I'm not sure if it's better, but it's the other way that came to mind.

Reply:
That would require our shell scripts to fork+exec `crudini` a bunch of times, which could noticeably slow down a no-op `provision`. `crudini` is itself a Python script that takes some 30 ms to run.